R/windsorize.R

```diff
@@ -1,10 +1,20 @@
-#' Windsorize
+#' Windsorizes data.
 #'
-#' Do some windsorization.
+#' @param x A numeric vector containing the data.
+#' @param p A numeric value denoting the a probability
+#' @return A numeric vector containing the windsorized data, i.e. extreme values are replaced by the quantiles derived from \code{p}.
+#' @examples
+#' windsorize(c(92 , 19 , 101 , 58 , 1053 , 91 , 26 , 78 , 10 , 13 , -40 , 101 , 86 , 85 , 15 , 89 , 89 , 28 , -5 , 41))
 #' @export
 windsorize <- function(x, p = .90) {
-  q <- quantile(x, p)
-  x[x >= q] <- q
+  if(length(x)==0){
+    stop("x must be of positive length")
+  }else if(sum(is.na(x))>0){
+    stop("x must not contain any NA")
```

Comment:
Does the function have to return an error if the vector contains `NA`'s? 
The specifications says it has to return an error if the vector contains ONLY `NA`'s. 
What could be the workaround?